### PR TITLE
Install compile docs

### PR DESCRIPTION
**Pull request by JhanSrbinovsky**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

I'm using VS locally to verify the formatting now, so it should have less errors than before. 

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix
- [X] New or updated documentation

## Checklist

- [ X] The new content is accessible and located in the appropriate section.
- [ X] I have checked that links are valid and point to the intended content.
- [ X] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

### CODE REVIEW

File: documentation/docs/user_guide/tutorial/compilation.md [hunk 1]
```diff
@@ -1,2 +1,89 @@
...
+
+The hidden `.tmp/` directory is really only needed for technical reasons and you should never need to go 
+into the `.tmp/` directory. However, if for some reason you want the **.o** object files created by the compiler, they persist in this directory. Alternatively, it is sometimes useful to verify that the files you want to compile are actually making it into this directory. This is particuarly relevant if you are adding new files to CABLE. 
+
+???+ warning "Do not change files under `.tmp/`"
+    If you change the files in `.tmp/` directly, the next build will not pick up these changes and will overwrite them.
+
+One of the features of the build process is that only source files which are
+modified are re-built, followed by their dependents. This is possible because the .''tmp'' directory is
+overwritten by the build script, preserving timestamps of the source files from their original location.
```
Comment (ccarouge) [inline documentation/docs/user_guide/tutorial/compilation.md:81]: I believe this is currently broken and one needs to remove the .tmp/ directory each time and reset the compilation to start from zero. Is that really so?

**Review by ccarouge — CHANGES_REQUESTED** [review]
I think the last part of the documentation is incorrect right now. Need confirmation though.

I have also extensively edited and committed my edits so @JhanSrbinovsky and @rkutteh , please have a look and don't hesitate to edit further through a review or via submitting further commits.

File: documentation/docs/user_guide/tutorial/compilation.md (hunk 1 above)
Comment (JhanSrbinovsky) [reply documentation/docs/user_guide/tutorial/compilation.md:81]: hmmm not sure what you mean. I can build, and continually rebuild and it only recompiles the file I'm editing it also rebuilds files that depend on it so they pick up the changes).  I only rm -fr .tmp etc when I move between mph and ifort. I actually do it at critical junctures as well just to be ultra cautious.

Comment (ccarouge) [reply documentation/docs/user_guide/tutorial/compilation.md:81]: OK. That's what I thought. It's only needed if changing between serial and parallel compilation. I'll add a warning about this in the doc.

**Review by ccarouge — APPROVED** [review]

**Comment by ccarouge** [comment]
@rkutteh do you want to review this or happy for me to merge into main at this point?

**Comment by ccarouge** [comment]
@JhanSrbinovsky you know you closed it without merging it in? You want to bin it all or that was a mistake?

**Review by rkutteh — APPROVED** [review]
Happy with it for now but will probably have something to say as more material goes in.

**Comment by JhanSrbinovsky** [comment]
merged on my end

**Pull request merged**